selfhost/typechecker.jakt

```diff
@@ -964,6 +966,69 @@ class CheckedProgram {
     public function get_bits(this, anon type_id: TypeId) => .get_type(type_id).get_bits()
 
     public function is_signed(this, anon type_id: TypeId) => .get_type(type_id).is_signed()
+
+    public function find_struct_in_scope(this, scope_id: ScopeId, name: String) throws -> StructId? {
```

Comment:
I am open to suggestions for naming. But having this function separately is nice because it handles all the necessary verifications if the type is weak ptr.